coffee_plugin/src/plugin/state.rs

```diff
@@ -41,6 +41,7 @@ impl State {
         self.args.clone().unwrap()
     }
 
+    #[allow(clippy::await_holding_lock)]
```

Comment:
mh wait this is dangerus, what is clippy telling you?

Reply:
This is the warning we discussed earlier over discord
the warning is 
```bash
warning: this `MutexGuard` is held across an `await` point
  --> coffee_plugin/src/plugin/state.rs:45:9
   |
45 | /         self.coffee()
46 | |             .lock()
47 | |             .unwrap()
   | |_____________________^
   |
   = help: consider using an async-aware `Mutex` type or ensuring the `MutexGuard` is dropped before calling await
note: these are all the `await` points this lock is held through
  --> coffee_plugin/src/plugin/state.rs:45:9
   |
45 | /         self.coffee()
46 | |             .lock()
47 | |             .unwrap()
48 | |             .setup(&self.args.clone().unwrap().conf)
49 | |             .await?;
   | |____________________^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#await_holding_lock
```
I couldn't solve it because it's used in `coffee_plugin/src/plugin/plugin_mod.rs` for example in regular synchronous functions so we can't use await  and we can't change them to `async` functions because they are functions in traits.